src/mon/OSDMonitor.cc

```diff
@@ -6660,7 +6660,10 @@ int OSDMonitor::prepare_pool_size(const unsigned pool_type,
       err = get_erasure_code(erasure_code_profile, &erasure_code, ss);
       if (err == 0) {
 	*size = erasure_code->get_chunk_count();
-	*min_size = std::min(erasure_code->get_data_chunk_count() + 1, *size);
```

Comment:
This explicitly allowed m=0.

Reply:
agreed. `m=0` renders an erasure codec RAID 0. i am not 100% sure if we should forbid this.

but i think we do need to add more sanity checks in `ErasureCode::parse()` to prevent wrong/bad `m`s.